featurebyte/query_graph/node/schema.py

```diff
@@ -412,3 +412,12 @@ class ColumnSpec(FeatureByteBaseModel):
     name: NameStr
     dtype: DBVarType
     timestamp_schema: Optional[TimestampSchema] = None
+
+    @model_validator(mode="after")
+    def _validate_string_timestamp_format_string(self) -> "ColumnSpec":
+        if self.dtype == DBVarType.VARCHAR and self.timestamp_schema is not None:
```

Comment:
should we also raise an error if `self.timestamp_schema` is None when the dtype is VARCHAR?

Reply:
Regular string columns will have `VARCHAR` dtype and `timestamp_schema` of None. We should raise this error only when the string column is a special datetime column, so it cannot be validated at the `ColumnSpec` level but at the table level.

This is currently being validated here in this PR: https://github.com/featurebyte/featurebyte/pull/2711/files#diff-dd1960347f0045df362d18aee725acbe91a1f6c6b01af116d58bfa91dcb5494fR99-R102